pkg/client/client.go

```diff
@@ -1106,7 +1105,6 @@ func (c *Client) CreateOrUpdateClusterRoleBinding(crb *rbacv1.ClusterRoleBinding
 	changed := reflect.DeepEqual(crb.RoleRef, existing.RoleRef)
 	changed = changed || reflect.DeepEqual(crb.Subjects, existing.Subjects)
 	changed = changed || reflect.DeepEqual(crb.Labels, existing.Labels)
-	changed = changed || reflect.DeepEqual(crb.Annotations, existing.Annotations)
```

Comment:
I believe we have to be a little smarter. If we just ignore the annotations then it is fine, but I think we have to preserve the existing annotations of the object that was found, aka `crb.Annotations = existing.Annotations`.

Reply:
cc @sttts @p0lyn0mial for verification

Reply:
you could "merge" the annotation map instead of ignoring it.

please have a look at https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourceapply/rbac.go#L58  this is what we use in our operators.

Reply:
Either merging, or restrict deep equal to the annotation "subdomain" you own, e.g. monitoring.openshift.io. But you have allow anything else without wiping it, e.g. `important.customer.com/foo: bar`. It's allowed for every actor to put annotations.

Reply:
@p0lyn0mial in case of merging annotations, should we follow the same pattern you have of merging whole ObjectMeta? Is there any benefit to such approach?

Reply:
We don't merge the whole ObjectMeta, just only a few things
```
func EnsureObjectMeta(modified *bool, existing *metav1.ObjectMeta, required metav1.ObjectMeta) {
	SetStringIfSet(modified, &existing.Namespace, required.Namespace)
	SetStringIfSet(modified, &existing.Name, required.Name)
	MergeMap(modified, &existing.Labels, required.Labels)
	MergeMap(modified, &existing.Annotations, required.Annotations)
}
```